### ISSUE TITLE
JOSS review: functionality

### ISSUE THREAD

**Issue by ranocha** [henry2004y/Vlasiator.jl#116]
- [x] The statement "Vlasiator.jl achieves optimal serial performance for single file processing" in the paper needs to be confirmed and backed up by data.
- [x] Reading "This part has not been carefully tested so it might not work or just generate wrong results. Contributions from users are warmly welcomed!" in the documentation at https://henry2004y.github.io/Vlasiator.jl/stable/manual/#Computing-derived-quantities does not make me feel good when trying to use this package. Maybe some additional testing would be good before publishing this software in JOSS? From the JOSS guidelines at https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort: "JOSS requires that software should be feature-complete (i.e., no half-baked solutions)".

Xref https://github.com/openjournals/joss-reviews/issues/4906

**Comment by henry2004y** [comment]
Update on the status about the derived variable calculations:

As of v0.9.32, we include tests for all listed variables. However, due to the intrinsic output options of Vlasiator, the same derived quantity may be calculated from different sources (e.g. scalar pressure from pressure tensor, or directly from VDF), or the required variables for computing others may have alternatives (e.g. current density from DCCRG B, or FSGRID B). Out current approach uses the most commonly used ones, whereas alternatives are left as future improvements. In terms of software development, this part is feature-complete; in terms of science, it is definitely not.

**Comment by henry2004y** [comment]
For the serial file I/O performance, we have separated the benchmarks (that was under the Log Section) to a [stand-alone section](https://henry2004y.github.io/Vlasiator.jl/dev/benchmark/) in the doc. This is used to highlight the improvement over the original Python package [Analysator](https://github.com/fmihpc/analysator). For array size larger than ~ 1MB, we can achieve ~ 3000 MB/s reading speed, which is quite close to what a typical SSD4 can get. We also realized that accurately estimating the theoretical maximum I/O performance on each platform is tricky, and is also beyond the scope of showing our current improvements over the Python implementation. Therefore, we do not show any comparison with respect to the system limits.
